internal/k8s/controllers/service_controller_reload.go

```diff
@@ -101,6 +108,9 @@ func (r *ServiceReconciler) reprocessAll(ctx context.Context, req ctrl.Request)
 		level.Info(r.Logger).Log("controller", "ServiceReconciler - reprocessAll", "event", "force service reload")
 		return ctrl.Result{}, errRetry
 	}
+	if r.FirstConfiguration.Load() {
+		r.FirstConfiguration.Store(false)
+	}
```

Comment:
nit: I think this is idempotent -`r.FirstConfiguration.Store(false)` should be enough (no need for the conditional)